yt/frontends/ramses/io.py

```diff
@@ -192,6 +213,9 @@ def _read_particle_fields(self, chunks, ptf, selector):
             for ax in "xyz":
                 if pn % ax not in field_list:
                     fields.append((ptype, pn % ax))
+            # attention : read only one chunk as its a global file, don't know wether this is the good way to do it
```

Comment:
ah yes, that would definitely solve our problem and it can be much more easy to later implement reader for any sort of file

Reply:
I can try to implement this if you want

Reply:
But actually, if you keep something like this : 

```
           tr.update(
                ph.reader(
                    fname, foffsets, data_types, subset, subs_fields, count=count
                )
            )
```

You are still updated the value of tr by appending the variable at the end, iterating over the chunks concerned. 

To keep using this method,  you need to check that the particle you are adding are well in the subset when you read it in the csv file. Thus checking with a few `if` the position of the particle. I don't know if it is fast (but here the files are less than 1e3 sinks so not too fast).

A 2nd method would be to read once for all the data of the csv file, in this case, you need to avoid to go in the iteration line 240 in `_read_particle_fields` : 
```
 for chunk in chunks:
            for subset in chunk.objs:
```

In this case, the class would store all the sink produced, independently of their position. 
It requires to either write some condition in  `_read_particle_fields` before going onto the iteration. 
Maybe, we can implement this with `_read_particle_data_file` method ?

Reply:
It might be difficult to check whether a particle is in the current subset though.

I think we have two options here that avoid having code specific to `sinks` here:
- we update `_read_particle_fields` so that it returns a boolean that informs whether all the data has been read (`sink` case) or not (all other cases).
- we update the particle handlers so that they carry a flag that says "all data is in one file, please only read one chunk".

At the end of the day, the reason for this (slight) mess is that the index assumes that the data is spread over `Ncpu` different domains, each of which forms one chunk. The best solution would require having a second index I guess, but that would be a massive change, especially for a simple case like this one.

Reply:
- we update `_read_particle_fields` so that it returns a boolean that informs whether all the data has been read (sink case) or not (all other cases).
    - in that case, we need to modify deeply the class `BaseIOHandler`, as we modify the output of `_read_particle_fields` no ?
    
- we update the particle handlers so that they carry a flag that says "all data is in one file, please only read one chunk".
    - yes, that could be the simplest idea no ?

what I wanted to say is that in the `BaseIOHandler` there is a `_read_particle_data_file`. I don't get it but there is a `raise NotImplementedError`, whereas it is not implemented in the `IOHandlerRAMSES` class. Maybe we can implement a method for data_file like this. So when reading the csv data, the called function is this one and not the former `_read_particle_fields`. The ideal, would be a function setted as an attribute of the particle class, but that might require to modify the class `BaseIOHandler` I think.

Reply:
You may be right. I don't remember the inheritance of `BaseIOHandler`. In that case, updating the particle handlers (they're specific to RAMSES and already encode all the specifics of each particle output file anyway) seems the best route.